pytorch_lightning/trainer/connectors/optimizer_connector.py

```diff
@@ -36,8 +36,11 @@ def update_learning_rates(self, interval: str, monitor_metrics=None):
             return
 
         for scheduler_idx, lr_scheduler in enumerate(self.trainer.lr_schedulers):
-            current_idx = self.trainer.batch_idx if interval == 'step' else self.trainer.current_epoch
-            current_idx += 1  # account for both batch and epoch starts from 0
+            current_idx = {'step': self.trainer.batch_idx,
+                           'epoch': self.trainer.current_epoch,
+                           'val': self.trainer.current_val_epoch}[interval]
```

Comment:
current_val_epoch is ambiguous as `val_check_interval<1`. `current_num_validation` or similar might be better.